spec/system/new_topic_list_spec.rb

```diff
@@ -0,0 +1,444 @@
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_inactive
+        expect(tabs_toggle.topics_tab).to be_active
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+
+        expect(current_url).to end_with("/new?s=topics")
+      end
+
+      it "shows only topics with new replies when the user switches to the Replies tab" do
+        visit("/new")
+        tabs_toggle.replies_tab.click
+
+        expect(topic_list).to have_topics(count: 3)
+        [new_reply, new_reply_in_category, new_reply_with_tag].each do |topic|
+          expect(topic_list).to have_topic(topic)
+        end
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_active
+        expect(tabs_toggle.topics_tab).to be_inactive
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+
+        expect(current_url).to end_with("/new?s=replies")
+      end
+
+      it "strips out the s (scope) query params when switching back to the All tab from any of the other tabs" do
+        visit("/new")
+        tabs_toggle.replies_tab.click
+
+        expect(tabs_toggle.all_tab).to be_inactive
+        expect(tabs_toggle.replies_tab).to be_active
+        try_until_success { expect(current_url).to end_with("/new?s=replies") }
+
+        tabs_toggle.all_tab.click
+
+        expect(tabs_toggle.all_tab).to be_active
+        expect(tabs_toggle.replies_tab).to be_inactive
+        try_until_success { expect(current_url).to end_with("/new") }
+      end
+
+      it "live-updates the counts shown on the tabs" do
+        visit("/new")
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+
+        TopicUser.update_last_read(user, new_reply_in_category.id, 2, 1, 1)
+
+        try_until_success do
+          expect(tabs_toggle.all_tab).to have_count(5)
+          expect(tabs_toggle.replies_tab).to have_count(2)
+          expect(tabs_toggle.topics_tab).to have_count(3)
+        end
+
+        TopicUser.update_last_read(user, new_topic.id, 1, 1, 1)
+
+        try_until_success do
+          expect(tabs_toggle.all_tab).to have_count(4)
+          expect(tabs_toggle.replies_tab).to have_count(2)
+          expect(tabs_toggle.topics_tab).to have_count(2)
+        end
+      end
+
+      context "when the /new topic list is scoped to a category" do
+        it "shows new topics and replies in the category" do
+          visit("/c/#{category.slug}/#{category.id}/l/new")
+          expect(topic_list).to have_topics(count: 2)
+          [new_reply_in_category, new_topic_in_category].each do |topic|
+            expect(topic_list).to have_topic(topic)
+          end
```

Comment:
Small recommendation to just duplicate code here instead of using a loop so that the backtrace is more useful when the assertion fails.